src/MAGMA.jl

```diff
@@ -34,11 +49,26 @@ macro magmafunc_nb(function_name)
 	return Expr(:quote, Symbol("magma_get_", function_name, "_nb"))
 end
 
+macro magmafunc_generic(function_name, str)
+	return Expr(:quote, Symbol(function_name, str))
+end
+
 const magmaTypeTuple= (Float32, Float64, ComplexF32, ComplexF64)
```

Comment:
this is sufficient. you don't need to parse strings manually.

Reply:
> this is sufficient. you don't need to parse strings manually.

Ah cool. There is some improvisation made by me a long time ago for convenience. Thank you for your careful checking and improvement.